src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.handler.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationOperationStrategyMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationOperationStrategyMap) {
+        this.operationOperationStrategyMap = operationOperationStrategyMap;
+    }
+
+    @Override
+    public OperationHandler getOperation(FruitTransaction.Operation operation) {
```

Comment:
Consider adding a null check for the 'operation' parameter in the 'getOperation' method to handle the case where a non-existent operation is requested. This will ensure that your code is robust and can provide a more informative error message if an invalid operation is passed.